app/models/user.rb

```diff
@@ -25,80 +30,184 @@ def to_s
+    !(solr_doc[solr_role] & groups).blank?
   end
 
-  @groups_to_impersonate = nil
-  # Allow a repository admin to see the repository as if they had a different set of permissions
-  # @param [Array<String>] set of groups
+  # Allow a repository admin to see the repository with different permissions
+  # @param grps [Array<String>|String|nil] set of groups
   def set_groups_to_impersonate(grps)
-    @groups_to_impersonate = grps
+    if grps.blank? # blank is falsy values like: nil, '', []
+      # remove impersonation (see #groups below)
+      @role_cache = {}
+      @groups_to_impersonate = nil
+    elsif grps.is_a? Array
+      @role_cache = {}
+      @groups_to_impersonate = grps
+    elsif grps.is_a? String
+      @role_cache = {}
+      @groups_to_impersonate = [grps]
+    else
+      raise ArgumentError.new("Cannot accept #{grps.class} argument")
+    end
   end
 
   def groups
-    return @groups_to_impersonate if @groups_to_impersonate
-    perm_keys = ["sunetid:#{login}"]
-    return perm_keys unless webauth && webauth.privgroup.present?
-    perm_keys + webauth.privgroup.split(/\|/).collect { |g| "workgroup:#{g}" }
+    return @groups_to_impersonate unless @groups_to_impersonate.blank?
+    @groups ||= begin
+      perm_keys = ["sunetid:#{login}"]
+      return perm_keys unless webauth && webauth.privgroup.present?
+      perm_keys + webauth.privgroup.split(/\|/).map {|g| "workgroup:#{g}"}
+    end
   end
 
   # @return [Boolean] is the user a repository wide administrator
   def is_admin
-    !(groups & ADMIN_GROUPS).empty?
-  end
-
-  # @return [Boolean] is the user a repository wide viewer
-  def is_viewer
-    !(groups & VIEWER_GROUPS).empty?
+    !(groups & ADMIN_GROUPS).blank?
   end
 
   # @return [Boolean] is the user a repo wide manager
   def is_manager
-    !(groups & MANAGER_GROUPS).empty?
+    !(groups & MANAGER_GROUPS).blank?
+  end
+
+  # @return [Boolean] is the user a repository wide viewer
+  def is_viewer
+    !(groups & VIEWER_GROUPS).blank?
   end
 
   # The convention is for boolean is_XYZ methods to be interrogative (end in "?")
   alias_method :is_admin?,   :is_admin
-  alias_method :is_viewer?,  :is_viewer
   alias_method :is_manager?, :is_manager
+  alias_method :is_viewer?,  :is_viewer
 
   def login
     webauth ? webauth.login : sunetid
   end
 
-  ##
+  # Validate access permissions using Argo user roles. If that fails, check the
+  # APO role permissions.  If the object is an APO, first check permissions on
+  # the governing APO of this APO object. As a last resort, check the object
+  # roles when this object is an APO.
+  #
+  # At present, there are a few methods that all do the same thing;
+  # see all the `can_manage_*` methods in Dor::Governable, i.e.
+  # https://github.com/sul-dlss/dor-services/blob/develop/lib/dor/models/governable.rb
+  #
+  # The upshot of this is that it’s not yet clear how Argo should query a Dor
```

Comment:
actually, hound might be right, i think you're using an apostrophe instead of a single quote:  `it’s` instead of `it's`